### ISSUE TITLE
Maybe size_hint() should return the estimated remaining chunk number

### ISSUE THREAD

**Issue by LokyinZHAO** [nlfiedler/fastcdc-rs#26]
Hi there!👋

I looked into the code of `v2020::<FastCDC as Iterator>::size_hint()`, which returns the estimated total number of chunks. But the documentation for [Iterator::size_hint()](https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.size_hint) says:

`size_hint() returns the bounds on the remaining length of the iterator.`

So it may be better to return the estimated number of **remaining** chunks:

```rust
    fn size_hint(&self) -> (usize, Option<usize>) {
        // NOTE: This intentionally returns the upper bound for both `size_hint`
        // values, as the upper bound doesn't actually seem to get used by `std`
        // and using the actual lower bound is practically guaranteed to require
        // a second capacity growth.
        let upper_bound = self.remaining / self.min_size;
        (upper_bound, Some(upper_bound))
    }
```

I'd be happy to make a PR if it is resonable.❤️

**Comment by nlfiedler** [comment]
You are indeed correct, it would be more accurate to return the number of remaining chunks. I welcome your PR. Thanks.

**Comment by nlfiedler** [comment]
Fixed in commit 9deaa281387f75f5690e8b1f90813e9e09e8b670
